language/use-cases/document-qa/utils/matching_engine_utils.py

```diff
@@ -11,11 +11,11 @@
 
 
 class MatchingEngineUtils:
-    def __init__(self, project_id: str, region: str, index_name: str):
+    def __init__(self, project_id: str, region: str, index_name: str, index_endpoint_name: str):
         self.project_id = project_id
         self.region = region
         self.index_name = index_name
-        self.index_endpoint_name = f"{self.index_name}-endpoint"
+        self.index_endpoint_name = index_endpoint_name
```

Comment:
To avoid breaking changes, can we make index_endpoint_name as optional and then default to `{index_name}-endpoint`.

Reply:
I made this change so this PR can be merged.